src/sagemaker/deserializers.py

```diff
@@ -60,3 +62,34 @@ def deserialize(self, data, content_type):
             return data.read()
         finally:
             data.close()
+
+
+class CSVDeserializer(BaseDeserializer):
+    """Deserialize a stream of bytes into a list of lists."""
+
+    ACCEPT = "test/csv"
+
+    def __init__(self, encoding="UTF-8"):
+        """Initialize the string encoding.
+
+        Args:
+            encoding (str): The string encoding to use (default: "UTF-8").
+        """
+        self.encoding = encoding
+
+    def deserialize(self, data, content_type):
+        """Deserialize data from an inference endpoint into a list of lists.
+
+        Args:
+            data (object): Data to be deserialized.
+            content_type (str): The MIME type of the data.
```

Comment:
unused arg, here and elsewhere

Reply:
Yeah. I've kept the content_type parameter in so that BaseDeserializer and all of it's subclasses have a consistent function signature

Reply:
NumpyDeserializer needs to know the content type to properly deserialize a response. The content type would also be required to deserialize JSON lines and JSON-LD formats. Presumably, some custom user deserializers may also need the content-type of the returned data to conditionally deserialize data.

To your second point, I don't think there's any case where the user would directly use a deserializer. The deserializer is used in the Predict class at this line: https://github.com/aws/sagemaker-python-sdk/blob/zwei/src/sagemaker/predictor.py#L129. A content-type is always passed as the second argument.